src/Validator/ConstraintsSetFactory/Field/IntegerFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AssoConnect\ValidatorBundle\Validator\ConstraintsSetFactory\Field;
+
+use AssoConnect\ValidatorBundle\Exception\UnsupportedIntegerTypeException;
+use Doctrine\DBAL\Types\Types;
+use Symfony\Component\Validator\Constraints\GreaterThanOrEqual;
+use Symfony\Component\Validator\Constraints\LessThanOrEqual;
+use Symfony\Component\Validator\Constraints\Type;
+
+/**
+ * @link https://dev.mysql.com/doc/refman/8.0/en/integer-types.html
+ */
+class IntegerFactory implements FieldConstraintsSetFactoryInterface
+{
+    public function supports(string $type): bool
+    {
+        return in_array(
+            $type,
+            [
+                Types::SMALLINT,
+                Types::INTEGER,
+                Types::BIGINT,
+            ],
+            true
+        );
+    }
+
+    public function getConstraints(array $fieldMapping): array
+    {
+        $isSigned = !isset($fieldMapping['options']['unsigned']) || false === $fieldMapping['options']['unsigned'];
+
+        return [
+            new Type('integer'),
+            new GreaterThanOrEqual($this->getMinValue($fieldMapping['type'], $isSigned)),
+            new LessThanOrEqual($this->getMaxValue($fieldMapping['type'], $isSigned)),
+
+        ];
+    }
+
+    /** @return int|float */
+    private function getMinValue(string $type, bool $isSigned)
+    {
+        if ($isSigned) {
+            switch ($type) {
+                case Types::SMALLINT:
+                    return -32768;
+                case Types::INTEGER:
+                    return -2147483648;
+                case Types::BIGINT:
+                    return 9223372036854775808;
```

Comment:
It seems to missing the minus to be a negative value. We probably improve this by refactoring these values to constants.